Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegateTests.swift

```diff
@@ -8,12 +8,21 @@ import XCTest
 @testable import Datadog
 
 class DDURLSessionDelegateTests: XCTestCase {
-    private let interceptor = URLSessionInterceptorMock()
-    private lazy var delegate: DDURLSessionDelegate = {
-        let delegate = DDURLSessionDelegate()
-        delegate.interceptor = interceptor
-        return delegate
-    }()
+    private let interceptor = URLSessionInterceptorMock.mockAny()
+    private lazy var delegate = DDURLSessionDelegate()
+
+    private var currentAutoInstrumentation: URLSessionAutoInstrumentation?
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        currentAutoInstrumentation = URLSessionAutoInstrumentation.instance
+        URLSessionAutoInstrumentation.instance = URLSessionAutoInstrumentation(interceptor: interceptor)
+    }
+
+    override func tearDownWithError() throws {
+        URLSessionAutoInstrumentation.instance = currentAutoInstrumentation
```

Comment:
We shouldn't need to store the previous instance (`currentAutoInstrumentation`). Each test should start with a clean state and reset it after completion, otherwise it will introduce flakiness. This should be enough:
```swift
URLSessionAutoInstrumentation.instance = nil
```